dashboard/app/models/lesson.rb

```diff
@@ -94,7 +95,7 @@ def self.add_lessons(script, lesson_group, raw_lessons, counters, new_suffix, ed
         lockable: !!raw_lesson[:lockable],
         has_lesson_plan: !!raw_lesson[:has_lesson_plan],
         visible_after: raw_lesson[:visible_after],
-        relative_position: !!raw_lesson[:lockable] ? (counters.lockable_count += 1) : (counters.non_lockable_count += 1)
+        relative_position: !!raw_lesson[:has_lesson_plan] || !raw_lesson[:lockable] ? (counters.lesson_count += 1) : (counters.survey_count += 1)
```

Comment:
nit: would it be easier to reverse this ternary? I personally feel like it's easier to check for the simplest case (what lessons are surveys) than the three other cases. But happy to be overruled here

Reply:
this does seem a bit hard to follow. I would suggest something like:
```
use_survey_url = ...
```
and then
```
relative_position: use_survey_url ? ... : ...
```
alternately, `use_survey_url` could be extracted as a method.